src/backend/distributed/commands/table.c

```diff
@@ -282,6 +282,15 @@ PostprocessCreateTableStmtForeignKeys(CreateStmt *createStatement)
 	bool missingOk = false;
 	Oid relationId = RangeVarGetRelid(createStatement->relation, NoLock, missingOk);
 
+	if (ShouldCreateTenantTable(relationId))
+	{
+		/*
+		 * Avoid unnecessarily adding the table into metadata if we will
```

Comment:
I kind of wonder whether this function should deal with CreateStmt more broadly.

It's a bit weird that we have auto-convert-on-CREATE-TABLE logic dispersed across different files